devtools/x/src/lint/guppy.rs

```diff
@@ -146,3 +146,51 @@ impl PackageLinter for CrateNamesPaths {
         Ok(RunStatus::Executed)
     }
 }
+
+/// Ensure libra-workspace-hack is a dependency
+#[derive(Debug)]
+pub struct WorkspaceHack;
+
+impl Linter for WorkspaceHack {
+    fn name(&self) -> &'static str {
+        "workspace-hack"
+    }
+}
+
+impl PackageLinter for WorkspaceHack {
+    fn run<'l>(
+        &self,
+        ctx: &PackageContext<'l>,
+        out: &mut LintFormatter<'l, '_>,
+    ) -> Result<RunStatus<'l>> {
+        let package_id = ctx.metadata().id();
+        let pkg_graph = ctx.project_ctx().package_graph().expect("can't find package graph");
+        let workspace_hack_id = pkg_graph.packages().find_map(|metadata| {
+            if metadata.name() == "libra-workspace-hack" {
+                return Some(metadata.id());
+            }
+            None
+        }).expect("can't find libra-workspace-hack package");
+
+        // libra-workspace-hack does not need to depend on itself
+        if package_id == workspace_hack_id {
+            return Ok(RunStatus::Executed);
+        }
+
+        if let Some(links) = pkg_graph.dep_links(package_id) {
+            let mut has_links = false;
```

Comment:
Why is `has_links` necessary?

Reply:
If there are no deps at all, there's no need to add workspace-hack. So we emit the lint only if it has deps *and* workspace-hack is not one of them.

Reply:
I tried for a while to find some version of find/map/filter, etc that would do this, but I was not successful.